data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/S3SinkConfig.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink;
+
+import org.opensearch.dataprepper.model.configuration.PluginModel;
+import org.opensearch.dataprepper.plugins.sink.configuration.AwsAuthenticationOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.ObjectOptions;
+import org.opensearch.dataprepper.plugins.sink.configuration.ThresholdOptions;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+
+/**
+ * An implementation class of s3 sink configuration
+ */
+public class S3SinkConfig {
+
+    private static final String DEFAULT_STORAGE = "in_memory";
+    private static final int DEFAULT_CONNECTION_RETRIES = 5;
+    private static final int DEFAULT_UPLOAD_RETRIES = 5;
+
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsAuthenticationOptions awsAuthenticationOptions;
+
+    @JsonProperty("threshold")
+    @NotNull
+    private ThresholdOptions thresholdOptions;
+
+    @JsonProperty("object")
+    @NotNull
+    private ObjectOptions objectOptions;
+
+    @JsonProperty("codec")
+    @NotNull
+    private PluginModel codec;
+
+    @JsonProperty("temporary_storage")

```

Comment:
I wonder if there is a better name here. Perhaps `buffer_type`? Or `event_buffer`?

Reply:
Updated the code with `buffer_type`